oxia/async_client_impl.go

```diff
@@ -212,10 +212,10 @@ func (c *clientImpl) List(ctx context.Context, minKeyInclusive string, maxKeyExc
 	ch := make(chan ListResult)
 	wg := common.NewWaitGroup(len(shardIds))
 	for _, shardId := range shardIds {
-		shardIdPtr := &shardId
+		shardIdPtr := shardId
 		go func() {
 			request := &proto.ListRequest{
-				ShardId:        shardIdPtr,
+				ShardId:        &shardIdPtr,
```

Comment:
This looks like it was an actual bug, right?

Reply:
Old implement is an bug（before [go1.22 and go1.21](https://github.com/golang/go/wiki/LoopvarExperiment)）